### DISCUSSION TITLE
using just DocExplorer

### DISCUSSION THREAD

**Discussion by jennifer-hiller** [graphql/graphiql#2227] [Ideas]
Is there a way I can just embed the DocExplorer into the page without including the entire GraphiQL interface?

**Comment by acao** [comment]
yes!

glad you asked here.

```ts
import { DocExplorer } from "graphiql"
```

In the near future, we hope to export it from a library of graphiql react components called `@graphiql/react`

**Reply by jennifer-hiller** [reply]
thanks! Is there a way I can pass the fetcher to it? I've tried `<DocExplorer fetcher={fetcher} /> ` but I'm not sure if it's being applied.

**Reply by jennifer-hiller** [reply]
Also, I'm actually using the library in an ES5 application, so I can't use import. I'm sourcing the library directly from the minified file. It doesn't seem to be exposing the DocExplorer by itself, but I'd like to change it so it does. Is there a simple way to do this?
`ReactDOM.render(React.createElement(GraphiQL.DocExplorer, {...}, containerElement)`
would be nice.

**Reply by acao** [reply]
The `DocExplorer` component requires a graphql schema instance as a `schema` prop. You can fetch a schema using fetch() and then use graphql-js buildClientSchema(). Hopefully can make an example of this soon.

As far as accomplishing this with the CDN bundle however, I’m not sure that it can be done. It would be better to have a higher level abstraction for these cases where JSX cannot be transpiled
